### PR TITLE
fix: grant pgsodium functions to service_role

### PR DESCRIPTION
**Pull request by soedirgo**
Using supautils's extension custom scripts. The script will be run on `create extension pgsodium`.

### CODE REVIEW

File: migrations/schema.sql [hunk 1]
```diff
@@ -510,25 +510,6 @@ END;
 $$;
 
 
---
--- Name: key_encrypt_secret(); Type: FUNCTION; Schema: pgsodium; Owner: -
---
-
-CREATE FUNCTION pgsodium.key_encrypt_secret() RETURNS trigger
```
Comment (soedirgo) [inline migrations/schema.sql:517]: The `schema` test has been failing for a while, so I guess this just reflects the latest pgsodium version.

File: migrations/schema.sql [hunk 2]
```diff
@@ -754,35 +735,6 @@ CREATE TABLE auth.users (
 COMMENT ON TABLE auth.users IS 'Auth: Stores user login data within a secure schema.';
 
 
---
--- Name: decrypted_key; Type: VIEW; Schema: pgsodium; Owner: -
---
-
-CREATE VIEW pgsodium.decrypted_key AS
```
Comment (soedirgo) [inline migrations/schema.sql:761]: @michelp do you know why this got dropped now? If I recreate `pgsodium` it gets dumped again.

File: migrations/schema.sql (hunk 1 above)
Comment (soedirgo) [reply migrations/schema.sql:517]: https://github.com/michelp/pgsodium/blob/060e27fc93a53999d94c58170627a443f0eb8d0c/sql/pgsodium--3.1.0--3.1.1.sql#L178

File: migrations/schema.sql (hunk 2 above)
Comment (soedirgo) [reply migrations/schema.sql:761]: I think I got it - this was fixed in pgsodium 3.1.1 but now [this issue](https://github.com/supabase/postgres/blob/00837a237134dc0cb7ba3d48fa96b67cc55da2a3/migrations/db/migrations/20221207154255_create_pgsodium_and_vault.sql#L13-L17) is back, so we need to revert the migration.

Comment (soedirgo) [reply migrations/schema.sql:761]: (I'll hold off on reverting it until Michel's done with the pgsodium fixes)

**Review by pcnc — APPROVED** [review]

**Comment by pcnc** [comment]
Looks good to me
@soedirgo is this ready to merge and test on staging, or are we waiting on feedback from Michel beforehand?

**Comment by soedirgo** [comment]
I guess we can merge and test this on staging and ask him later - one less thing to fix at once

**Pull request merged**